notesapi/v1/tests/test_views.py

```diff
@@ -546,6 +567,93 @@ def test_search_course(self):
         self.assertEqual(results['total'], 1)
         self.assertEqual(results['rows'][0]['text'], u'Third note')
 
+    def test_search_tag(self):
+        """
+        Tests searching for tags
+        """
+        self._create_annotation(text=u'First note', tags=[u'foo', u'bar'])
+        self._create_annotation(text=u'Another one', tags=[u'bar'])
+        self._create_annotation(text=u'A third note', tags=[u'bar', u'baz'])
+        self._create_annotation(text=u'One final note', tags=[])
+
+        results = self._get_search_results(text='Foo')
+        self.assertEqual(results['total'], 1)
+        self.assertEqual(results['rows'][0]['text'], 'First note')
+
+        results = self._get_search_results(text='bar')
+        self.assertEqual(results['total'], 3)
```

Comment:
I think you should assert that each of the notes are what you expect them to be, since in this case you get multiple results; so far we have only inspected the notes when we get one result. You could write a helper method and then call it as well from `test_search_ordering_in_es()` and `_search_tag_or_text()`.